src/java.base/share/native/libjli/java.c

```diff
@@ -1546,9 +1382,76 @@ ParseArguments(int *pargc, char ***pargv,
 
     *pmode = mode;
 
+    if (!headless) {
+        char *jar_path = NULL;
+        if (mode == LM_JAR) {
+            jar_path = *pwhat;
+        }
+        // Not in headless mode. We now set a couple of env variables that
+        // will be used later by ShowSplashScreen().
+        SetupSplashScreenEnvVars(splash_file_path, jar_path);
+    }
+
     return JNI_TRUE;
 }
 
+static void
+SetupSplashScreenEnvVars(const char *splash_file_path, // "-splash:" option value (may be NULL)
+                         char *jar_path // the jar file being launched (may be NULL)
+) {
```

Comment:
I think this would be a bit cleaner if you put comment on the function to document the two args. That would remove the // comments that make this hard to read right now.

Reply:
Done, I've now moved those comments as function documentation.